pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -386,8 +392,13 @@ func (og *operationGenerator) GenerateDetachVolumeFunc(
 		if err != nil {
 			return volumetypes.GeneratedOperations{}, volumeToDetach.GenerateErrorDetailed("DetachVolume.SplitUniqueName failed", err)
 		}
-		attachableVolumePlugin, err = og.volumePluginMgr.FindAttachablePluginByName(pluginName)
-		if err != nil {
+
+		var ok bool
+		attachableVolumePlugin, ok, err = og.volumePluginMgr.FindAttachablePluginByName(pluginName)
+		if !ok {
+			if err == nil {
+				err = fmt.Errorf("plugin %s isn't a AttachableVolumePlugin", pluginName)
+			}
```

Comment:
I think it is simpler to just check for `nil`  `attachableVolumePlugin` in the code rather than introducing the boolean return.